### DISCUSSION TITLE
"net_lwip_webserver (RNDIS) example" and FreeRTOS

### DISCUSSION THREAD

**Discussion by gionatha** [hathach/tinyusb#871] [Q&A]
I tried the "net_lwip_webserver" sample project on an NXP card (imxrt 1064) and it works perfectly.
I also tested the two freertos projects (cdc_msc_freertos and hid_composite_freertos) without having any problems.
I then tried to merge the projects so that we have net_lwip_webserver + FreeRTOS. I did not think I would have difficulties given the starting examples, but the new project does not work correctly.
The RNDIS device is correctly and quickly detected by Windows 10, the ping to the addresses 192.168.7.1 and 192.168.7.2 work fine, but I cannot open the page at the address 192.168.7.1 from the browser (with the example no-FreeRTOS is very fast).
I have made many attempts to configure the lwip and FreeRTOS libreiries but to no avail, I don't really understand where the problem is.
Can someone help me?
Thank you
Gionatha

**Comment by gionatha** [comment]
Hi all, sorry again.
I am still trying to use the "net_lwip_webserver" example with FreeRTOS, unfortunately with no luck. There are 3 actors involved: tinyusb, FreeRTOS and LwIP.
Libraries provide example with TinyUSB and LwIP (without FreeRTOS).
NXP has an example of a webserver with LwIP and FreeRTOS.
I tried both to start from the first and add FreeRTOS, and from the second by adding the LwIP library, unfortunately without results.
Could anyone please have a working example or can help me solve the problem? Thank you very much.
Gionatha

**Comment by hannnnn-l** [comment]
I successes last night on this. My tricks are: 1. keep tinyusb and lwip in seperate tasks; 2. replace the tud_task() on line 110 in net_lwip_webserver‘s main.c with a delay. It works for me.
